compparallel.go

```diff
@@ -0,0 +1,267 @@
+package routinghelpers
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/routing"
+)
+
+var _ routing.Routing = &Parallel{}
+
+type ComposableParallel struct {
+	routers []*ParallelRouter
+}
+
+func NewComposableParallel(routers []*ParallelRouter) *ComposableParallel {
+	return &ComposableParallel{
+		routers: routers,
+	}
+}
+
+func (r *ComposableParallel) Provide(ctx context.Context, cid cid.Cid, provide bool) error {
+	var wg sync.WaitGroup
+	errCh := make(chan error)
+	closeCh := make(chan bool)
+	for _, r := range r.routers {
+		r := r
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			select {
+			case <-ctx.Done():
+				if ctx.Err() != nil && !r.IgnoreError {
```

Comment:
If `ctx.Done()` is closed, then `ctx.Err()` will always be non-nil.

Reply:
fixed